### PR TITLE
(fix) - HMR and Typescript

### PR DESCRIPTION
**Pull request by JoviDeCroock**
On windows the exclude of the Babel.overrides doesn't work making transformations not apply correctly. We had multiple copies of the devServer present and some of them hot others not which make WDS confused.

I need to accurately test this PR a bit more, tested on Windows and it works now need to do it on a UNIX machine.

### CODE REVIEW

File: packages/cli/lib/lib/webpack/webpack-client-config.js [hunk 1]
```diff
@@ -36,14 +36,6 @@ async function clientConfig(env) {
 		polyfills: resolve(__dirname, './polyfills'),
 	};
 
-	if (!isProd) {
-		entry.bundle = [
```
Comment (JoviDeCroock) [inline packages/cli/lib/lib/webpack/webpack-client-config.js:40]: Removed the unneeded devServers

File: packages/cli/lib/lib/webpack/run-webpack.js [hunk 2]
```diff
@@ -73,7 +73,6 @@ async function devBuild(env) {
 
 		let c = Object.assign({}, config.devServer, {
 			stats: { colors: true },
-			hot: !env.refresh,
```
Comment (JoviDeCroock) [inline packages/cli/lib/lib/webpack/run-webpack.js:76]: Enabled the hot-server which we use as a base

File: packages/cli/lib/lib/babel-config.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ module.exports = function (env, options = {}) {
 		overrides: [
 			// Transforms to apply only to first-party code:
 			{
-				exclude: /(^|\/|\\)node_modules(\/|\/)/,
+				exclude: /node_modules/,
```
Comment (JoviDeCroock) [inline packages/cli/lib/lib/babel-config.js:39]: Windows compatible exclude

Comment (ForsakenHarmony) [reply packages/cli/lib/lib/babel-config.js:39]: I think the second `\/` at the end of the old regex was maybe supposed to be a `\\`, but this should be fine as well

File: packages/cli/lib/lib/webpack/webpack-client-config.js (hunk 1 above)
Comment (ForsakenHarmony) [reply packages/cli/lib/lib/webpack/webpack-client-config.js:40]: Does this automatically get added by the plugin / config option?

File: packages/cli/lib/lib/babel-config.js (hunk 3 above)
Comment (developit) [reply packages/cli/lib/lib/babel-config.js:39]: lol yeah I thought that was odd. This should work:

```suggestion
				exclude: /(^|\/|\\)node_modules(\/|\\)/,
```

Comment (developit) [reply packages/cli/lib/lib/babel-config.js:39]: Another option would be to use the match pattern:

```suggestion
				exclude: '**/node_modules/**',
```

File: packages/cli/lib/lib/webpack/webpack-client-config.js (hunk 1 above)
Comment (JoviDeCroock) [reply packages/cli/lib/lib/webpack/webpack-client-config.js:40]: This is part of the `DevServer.hot` option

File: packages/cli/lib/lib/webpack/webpack-client-config.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ const cleanFilename = (name) =>
 	);
 
 async function clientConfig(env) {
-	const { isProd, source, src, refresh, cwd /*, port? */ } = env;
+	const { isProd, source, src, cwd /*, port? */ } = env;
```
Comment (marvinhagemeister) [inline packages/cli/lib/lib/webpack/webpack-client-config.js:28]: We should probably remove any mentions of the `--refresh` flag in the Readme too👍

**Review by marvinhagemeister — APPROVED** [review]
This is fantastic! Outstanding debugging skills🙌

File: packages/cli/lib/lib/webpack/webpack-client-config.js (hunk 4 above)
Comment (JoviDeCroock) [reply packages/cli/lib/lib/webpack/webpack-client-config.js:28]: it still works, just removed from the `clientConfig` and now exclusively present in the `devConfig`

**Review by reznord — APPROVED** [review]

**Comment by Inviz** [comment]
Thank you for giving this issue your attention!

**Pull request merged**